instrumentation/opentelemetry-instrumentation-threading/src/opentelemetry/instrumentation/threading/__init__.py

```diff
@@ -0,0 +1,65 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=empty-docstring,no-value-for-parameter,no-member,no-name-in-module
+
+import threading  # pylint: disable=import-self
+from typing import Collection
+
+from opentelemetry import context, trace
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.threading.package import _instruments
+from opentelemetry.instrumentation.threading.version import __version__
+from opentelemetry.trace import (
+    get_current_span,
+    get_tracer,
+    get_tracer_provider,
+)
+
+
+class _InstrumentedThread(threading.Thread):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._parent_span = None
+
+    def start(self):
```

Comment:
If you did `self._parent_span = get_current_span()` in `__init__()`, the new thread would inherit the context of the thread that created the new thread object. Here, the new thread inherits the context of the thread that calls `start()`.

In the vast majority of cases they are probably the same caller though. Is this an intentional distinction @pridhi-arora ? Otherwise I agree with Diego that would be simpler

Reply:
But the current context may be different when one creates the thread object vs when it gets called. Take the following scenario.

```python
user_thread = threading.Thread(target=func, args=('hyd',))

with tracer.start_as_current_span("foo"):
  bar()
  user_thread.start()
```

It would be incorrect to use any span context other than "foo" right? We need this for that reason.